diff --git a/src/libs/mps_comm/machine.cpp b/src/libs/mps_comm/machine.cpp
index 2af1040..128caa8 100644
--- a/src/libs/mps_comm/machine.cpp
+++ b/src/libs/mps_comm/machine.cpp
@@ -180,7 +180,7 @@ Machine::~Machine() {
   
 
 void Machine::set_light(llsf_msgs::LightColor color, llsf_msgs::LightState state, unsigned short time) {
-  LightColor m_color;
+  LightColor m_color = LightColor::LIGHT_COLOR_RESET;
   switch (color) {
     case llsf_msgs::LightColor::RED:
       m_color = LightColor::LIGHT_COLOR_RED;
